core/src/test/java/de/danoeh/antennapod/core/storage/DbWriterTest.java

```diff
@@ -778,6 +778,34 @@ public void testMarkFeedRead() throws Exception {
             assertTrue(item.isPlayed());
         }
     }
+    
+    @Test
+    public void testRemoveAllNewFlags() throws Exception {
+        final int numItems = 10;
+        Feed feed = new Feed("url", null, "title");
+        feed.setItems(new ArrayList<>());
+        for (int i = 0; i < numItems; i++) {
+            FeedItem item = new FeedItem(0, "title " + i, "id " + i, "link " + i,
+                    new Date(), FeedItem.NEW, feed);
+            feed.getItems().add(item);
+        }
+
+        PodDBAdapter adapter = PodDBAdapter.getInstance();
+        adapter.open();
+        adapter.setCompleteFeed(feed);
+        adapter.close();
+
+        assertTrue(feed.getId() != 0);
+        for (FeedItem item : feed.getItems()) {
+            assertTrue(item.getId() != 0);
+        }
+
+        DBWriter.removeAllNewFlags();
+        List<FeedItem> loadedItems = DBReader.getFeedItemList(feed);
+        for (FeedItem item : loadedItems) {
+            assertFalse(item.isPlayed());
```

Comment:
This test succeeds even if the item is still `New`. You should instead test if it is no longer new.

Reply:
Thank you for pointing it out, just correct it to check if the feeditem is no longer new.